docs/source/contribute/new-data-source.rst

```diff
@@ -0,0 +1,89 @@
+Structured Data Source Integration
+====
+This document details steps invovled in adding a new structured data source integration in EvaDB.
+
+
+Example Data Source Integration In EvaDB
+----
+
+- `PostgreSQL <https://github.com/georgia-tech-db/evadb/tree/master/evadb/third_party/databases/postgres>`_
+
+
+Create Data Source Handler
+----
+
+1. Create a new directory at `evadb/third_party/databases/ <https://github.com/georgia-tech-db/evadb/tree/master/evadb/third_party/databases>`_
+~~~~
+
+.. note::
+
+   The directory name is also the engine name used in the `CREATE DATABASE mydb_source WITH ENGINE = "..."`. In this document, we use **mydb** as the example data source we want to integrate in EvaDB.
+
+The directory should contain three files:
+
+- __init__.py
+- requirements.txt
+- mydb_handler.py
+
+The *__init__.py* can contain copyright information. The *requirements.txt* contains the extra python libraries that need to be installed via pip for the mydb data source. 
+
+.. note:: 
+
+   EvaDB will only install a data source's specific dependency libraries when a connection to the data source is created by the user via, e.g., `CREATE DATABASE mydb_source WITH ENGINE = "mydb";`.
```

Comment:
@jiashenC This is what I have for requirements.txt. Could you help elaborate on that? Thanks!

Reply:
Oh, i see. I missed that part.